ignite/distributed/comp_models/native.py

```diff
@@ -0,0 +1,204 @@
+import os
+import subprocess
+from typing import Optional, Union
+
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+
+from ignite.distributed.comp_models.base import ComputationModel
+
+
+class _NativeDistModel(ComputationModel):
+    """Private class for PyTorch native distributed computation model.
+
+    Supported `backends <https://pytorch.org/docs/stable/distributed.html#backends>`_:
+
+    - NCCL
+    - GLOO
+    - MPI
+
+    In this implementation we assume the following mapping between backend and devices:
+
+    - NCCL <-> GPU
+    - GLOO <-> CPU
+    - MPI  <-> CPU
+
+    """
+
+    name = "native-dist"
+    available_backends = tuple(
+        name
+        for name in [dist.Backend.NCCL, dist.Backend.GLOO, dist.Backend.MPI]
+        if getattr(dist, "is_{}_available".format(name))
+    )
+
+    @staticmethod
+    def create_from_context() -> Optional["_NativeDistModel"]:
```

Comment:
Returned type should be `Optional[ComputationModel]` to respect the `ComputationModel. create_from_context` type hint.

Same for `create_from_backend` below.

Reply:
As type hints they are more precise

Reply:
Too influenced by c++ 😅